src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -89,41 +89,52 @@ dual c = mempty
          }
 
 -- Give the opposite initial state for the chosen signal
-bubbleInitialValue :: Eq a => a -> (a -> InitialValue) -> (a -> InitialValue)
-bubbleInitialValue s f y = if y == s then bubbleVal (f y) else f y
+bubbleInitialValue :: Eq a => [a] -> (a -> InitialValue) -> (a -> InitialValue)
+bubbleInitialValue s f y = if y `elem` s then bubbleVal (f y) else f y
   where
     bubbleVal (Defined v) = Defined (not v)
     bubbleVal x = x
 
 -- Perform the inversion for just transitions of the given signal
-toggleSpecific :: Eq a => a -> Transition a -> Transition a
+toggleSpecific :: Eq a => [a] -> Transition a -> Transition a
 toggleSpecific a t
-    | signal t == a = toggle t
-    | otherwise     = t
+    | signal t `elem` a = toggle t
+    | otherwise         = t
 
 -- For the selected signal, toggle the directions of all transitions for this.
-bubbleCausality :: Eq a => a -> Causality (Transition a)
-                        -> Causality (Transition a)
+bubbleCausality :: Eq a => [a] -> Causality (Transition a) -> Causality (Transition a)
 bubbleCausality s (Causality f t)
-    | signal t == s = Causality (map (toggleSpecific s) f) (toggle t)
-    | otherwise     = Causality (map (toggleSpecific s) f) t
+    | signal t `elem` s = Causality (map (toggleSpecific s) f) (toggle t)
+    | otherwise         = Causality (map (toggleSpecific s) f) t
 
 -- Invert the invariant transition of the selected signal.
-bubbleInvariant :: Eq a => a -> Invariant (Transition a)
-                        -> Invariant (Transition a)
+bubbleInvariant :: Eq a => [a] -> Invariant (Transition a) -> Invariant (Transition a)
 bubbleInvariant s (NeverAll es) = NeverAll (map (toggleSpecific s) es)
 
+-- Invert all initial states, causality transitions and invariant transitions
+-- of a single given signal in a given concept.
 bubble :: Eq a => a -> CircuitConcept a -> CircuitConcept a
-bubble s c = mempty
-             {
-                initial = bubbleInitialValue s (initial c),
-                arcs = fmap (bubbleCausality s) (arcs c),
-                interface = interface c,
-                invariant = fmap (bubbleInvariant s) (invariant c)
-             }
+bubble s c = bubbles [s] c
 
--- Signal-level concepts
+-- Invert all initial states, causality transitions and invariant transitions
+-- of a list of given signals in a given concept.
+bubbles :: Eq a => [a] -> CircuitConcept a -> CircuitConcept a
+bubbles s c = mempty
+    {
+       initial = bubbleInitialValue s (initial c),
+       arcs = fmap (bubbleCausality s) (arcs c),
+       interface = interface c,
+       invariant = fmap (bubbleInvariant s) (invariant c)
+    }
 
+-- Apply enable signal only for transitions of given signals in the concept
```

Comment:
I think in practice we almost always will apply `enable` only to a single output, so I'd prefer `enable` to take a single signal only. You can add `enables`, which sounds a bit odd, but is probably OK from the symmetry point of view :)